src/components/common/PageHeader/PageHeader.js

```diff
@@ -16,14 +16,27 @@ const Span = styled.span`
   }
 `;
 
-const PageHeader = ({
-  logOut,
-  user,
-  edits,
-  visible,
-  closeSideBar,
-  toggleSideBar
-}) => {
+const PageHeader = ({ logOut, user, edits }) => {
+  const [visible, setVisible] = useState(false);
+
+  const closeSideBar = () => {
+    if (visible) {
+      setVisible(!visible);
+    }
+  };
+
+  const toggleSideBar = () => {
+    setVisible(!visible);
+  };
+
+  React.useEffect(function setupListener() {
```

Comment:
![image](https://user-images.githubusercontent.com/47332187/65267693-d8d2ca80-db15-11e9-9707-9f6f5699de44.png)

Reply:
Didn't you forget the Dependencies Array?

Reply:
Hahahah. Well is so sorcery that I learnt that its behaviour is different when there is no array than when the array is empty 🤯 If you set console logs just before the event is added and another one just before the event is removed you'll see it.

Reply:
To see how it plays out after the console logs you need to open the panel and click in anywhere to close it, that's when it's triggered.